sdk/src/server-api/sc/api/plugins/TwoPlayerGameState.kt

```diff
@@ -20,40 +20,36 @@ abstract class TwoPlayerGameState<P : Player>(
     val players: List<P>
         get() = listOf(first, second)
     
+    /** Die Namen der beiden Spieler. */
+    val playerNames: Array<String>
+        get() = arrayOf(first.displayName, second.displayName)
```

Comment:
Any reason why this isn't a list like `players`, too?

Reply:
why would it? A list is more expensive to construct and in Kotlin almost all extension functions are available for both.

Reply:
1. `players` is called far more often but still a list
2. `playerNames` is rarely ever called
3. Arrays are mutable, potentially requiring more effort in syncing them between threads (this is hypothetical, I don't know how Kotlin handles this)
4. Since Arrays are mutable, `playerNames` appears to be something you can change even though you can't (and shouldn't)
5. If performance really is the issue, then consider initialising it lazily as the `playerNames` are not going to change. The extra bytes of storage are negligable

Reply:
(5) lazy init does not work because of xstream, we already had that issue a few times ;)
(1) yes, was just about to look at that as well
(3) and (4) arrays don't have any special thread safety nor mutability behavior. In Java the main difference is that the size of an Array is fixed, while a List can grow and expand (it is usually an ArrayList, so it uses an Array underneath anyways).
Fair, Kotlin adds semantics to make Lists immutable by default, but since these are instantiated for the sole use by the caller they might as well be mutable.

Reply:
(1) as for players being a list: they are parametrised, meaning that we can't use an Array as these can't use generics. I just tried.

Reply:
(1) Have you tried `Array<out P>`?

Reply:
5. you could try something like
```
lateinit var playerNames: Array<String>
  private set
  get() {
    if (!field.isInitialised()) 
      field = listOf(first.displayName, second.displayName)
    return field
  }

but then again, this is all just premature optimisation

Reply:
>  this is all just premature optimisation

exactly, so let's just go with Array